pandas/core/frame.py

```diff
@@ -3577,20 +3583,21 @@ def _get_value(self, index, col, takeable: bool = False) -> Scalar:
         series = self._get_item_cache(col)
         engine = self.index._engine
 
+        if isinstance(self.index, CategoricalIndex):
+            # Trying to use the engine fastpath may give incorrect results
+            #  if our categories are integers that dont match our codes
+            col = self.columns.get_loc(col)
+            index = self.index.get_loc(index)
+            return self._get_value(index, col, takeable=True)
+
         try:
             loc = engine.get_loc(index)
             return series._values[loc]
-        except KeyError:
-            # GH 20629
-            if self.index.nlevels > 1:
-                # partial indexing forbidden
-                raise
-
-        # we cannot handle direct indexing
-        # use positional
-        col = self.columns.get_loc(col)
-        index = self.index.get_loc(index)
-        return self._get_value(index, col, takeable=True)
+        except AttributeError:
```

Comment:
on the list to add?

Reply:
once this is merged ill push my branch that implements IntervalTree.get_loc, getting rid of the AttributeError